### DISCUSSION TITLE
Feature Request: Allow routes to be bound from namespaces other than `App\`.

### DISCUSSION THREAD

**Discussion by sbrow** [spatie/laravel-route-attributes#59] [Ideas]
It appears as though it's not currently possible to bind routes inside a custom Laravel package. This would be a great feature.

Perhaps the config file fomat could be updated to include namespaces along with the directories?

```php
<?php

return [
    /*
     *  Automatic registration of routes will only happen if this setting is `true`
     */
    'enabled' => true,

    /*
     * Controllers in these directories that have routing attributes
     * will automatically be registered.
     */
    'directories' => [
        'App\Http\Controllers' => app_path('Http/Controllers'),
        'User\Package\Http\Controllers => 'base_path('vendor/user/package/src/Http/Controllers'),
    ],

    /**
     * This middleware will be applied to all routes.
     */
    'middleware' => [
        \Illuminate\Routing\Middleware\SubstituteBindings::class
    ]
];

```

Thanks, and keep up the awesome work!

**Comment by paramientos** [comment]
Yes, I totally agree i am currently using nwidart module package and class fqdn name specifies wrong and the route binding is not possible.
The following method cannot resolve the class name properly (https://github.com/spatie/laravel-route-attributes/blob/master/src/RouteRegistrar.php#L86)
```
protected function fullQualifiedClassNameFromFile(SplFileInfo $file): string
    {
        $class = trim(Str::replaceFirst($this->basePath, '', $file->getRealPath()), DIRECTORY_SEPARATOR);
        $class = str_replace(
            [DIRECTORY_SEPARATOR, 'App\\'],
            ['\\', app()->getNamespace()],
            ucfirst(Str::replaceLast('.php', '', $class))
        );

        return $this->rootNamespace.$class;
    }
```

**Comment by freekmurze** [comment]
Feel free to send a PR with tests and updated readme for this.

**Reply by paramientos** [reply]
I pushed the feature but guess what I forgot: test. Sorry i am going to implement it asap

**Reply by danielsetreus** [reply]
Where did we end up with this? I have the same idea for a PR - but feels stupid to start development if it's already been done in some capacity.

**Reply by danielsetreus** [reply]
Fixed by #69
